app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/OffboardRouterActivityKt.kt

```diff
@@ -129,27 +127,24 @@ class OffboardRouterActivityKt : AppCompatActivity(),
                 waypoints.forEach { addWaypoint(it) }
             }.build()
 
-            session.requestRoutes(options)
+            offboardRouter?.getRoute(options, this@OffboardRouterActivityKt)
         }
     }
 
     /*
-     * DirectionSessions.RouteObserver
+     * Router.Callback
      */
 
-    override fun onRoutesChanged(routes: List<Route>) {
+    override fun onResponse(routes: List<Route>) {
         routes.firstOrNull()?.let {
             route = it.mapToDirectionsRoute()
             navigationMapRoute.addRoute(route)
         }
     }
 
-    override fun onRoutesRequested() {
-        MapboxLogger.d(Message("onRoutesRequested: navigation.getRoute()"))
-    }
-
-    override fun onRoutesRequestFailure(throwable: Throwable) {
-        MapboxLogger.e(Message("onRoutesRequestFailure: navigation.getRoute()"), throwable)
+    override fun onFailure(throwable: Throwable) {
+        Toast.makeText(this, "Error: " + throwable.message, Toast.LENGTH_LONG).show()
```

Comment:
String interpolation?

Reply:
👍 I just copied fro the Java example, we can fixup later.